### ISSUE TITLE
Failed Build openssl-3.0.5 for rbenv install 3.1.2

### ISSUE THREAD

**Issue by mannyc-dev** [rbenv/ruby-build#2050]
- I am on the latest version of Ubuntu 22.04.1
- Followed the instructions for the basic checkout git (also on a fresh install used the automated script)
- When running `rbenv install 3.1.2` an error message is received that the build failed because it is unable to make/install openssl-3.0.5. It seems openssl-3.0.5 is for the next version of Ubuntu (kinetic)
- After countless hours I was able to install ruby by downloading dependencies that are required to install openssl-3.0.5 and ruby. The below command will install the dependencies needed.

`sudo apt install libssl-dev libz-dev build-essential checkinstall zlib1y-dev`

-Perhaps the instructions or the script needs to be re-written?

**Comment by mislav** [comment]
(Transferred your issue to the ruby-build repository)

Hi, sorry you've had trouble, and thanks for sharing your solution steps!

If you had seen this earlier, would it have helped? https://github.com/rbenv/ruby-build/wiki#ubuntudebianmint
We could try to link to it more prominently from the `rbenv/rbenv` repository.

**Comment by mannyc-dev** [comment]
Yes! That would have helped. I didn't see the checkinstall. It seems like openssl-3.0.5 requires it.

**Comment by JeffCohen** [comment]
(I hope it's ok to ask here but happy to move this to Discussion if that's preferred)

I regret posting because I know this has been a common issue, but I feel like I've read everything I can regarding workarounds and fear I'm missing something obvious. I'm also trying to install 3.1.2 on a new Ubuntu 22.04.1 LTS machine.  I made sure to `brew upgrade ruby-build` and also installed all of the extra extensions mentioned here and in the rbenv post that @mislav provided above.  I continue to get this error (I can post the full verbose log) if that's helpful.

My understanding is that ruby-build would try to grab the older version of openssl if needed, but I'm not seeing that happen (and maybe it doesn't need to in this case anyway?)

```
The Ruby openssl extension was not compiled.
ERROR: Ruby install aborted due to missing extensions
Try running `apt-get install -y libssl-dev` to fetch missing dependencies.

Configure options used:
  --prefix=/home/deploy/.rbenv/versions/3.1.2
  --enable-shared
  --with-readline-dir=/home/linuxbrew/.linuxbrew/opt/readline
  LDFLAGS=-L/home/deploy/.rbenv/versions/3.1.2/lib
  CPPFLAGS=-I/home/deploy/.rbenv/versions/3.1.2/include

BUILD FAILED (Ubuntu 22.04 using ruby-build 20220910.1)

Inspect or clean up the working tree at /tmp/ruby-build.20220923165340.2544.hqIt35
Results logged to /tmp/ruby-build.20220923165340.2544.log

Last 10 log lines:
The Ruby openssl extension was not compiled.
ERROR: Ruby install aborted due to missing extensions
Try running `apt-get install -y libssl-dev` to fetch missing dependencies.

Configure options used:
  --prefix=/home/deploy/.rbenv/versions/3.1.2
  --enable-shared
  --with-readline-dir=/home/linuxbrew/.linuxbrew/opt/readline
  LDFLAGS=-L/home/deploy/.rbenv/versions/3.1.2/lib
  CPPFLAGS=-I/home/deploy/.rbenv/versions/3.1.2/include
```

(Of course I've run `apt-get install -y libssl-dev` just to be triple-sure and still get this error. )

Any thoughts?

**Comment by mislav** [comment]
Could you paste the contents of the whole log file as a gist? Thanks

**Comment by JeffCohen** [comment]
Thanks mislav!  Here it is:

https://gist.github.com/JeffCohen/e701b7cedb08db06890c31cd79d91018

**Comment by mislav** [comment]
@JeffCohen Strange! This Ruby version should have been linked to your system OpenSSL, but for some reason it wasn't.

Can you check your environment variables to see if there is anything that could have affected compilation? See `env`

@hsbt Do you have any ideas?

**Comment by JeffCohen** [comment]
Here's my env vars: https://gist.github.com/JeffCohen/85339684ce96131d77318178cc986f03

Just in case it's important: this is my very first time using `brew` on Linux, honestly until this project I think brew was only for mac.  I followed the standard install instructions and did nothing custom.  I used brew to install rbenv.  

But I used `apt` to install all the other dependencies per various instructions.  Maybe that's not ok because of certain path lookup rules (or something)?

**Comment by mislav** [comment]
Thanks for the share. I don't see anything odd in your env.

It shouldn't be a problem that you used Homebrew for anything. You should still use apt for system packages such as `libssl-dev`. And since you've installed OpenSSL headers, Ruby should have linked to them automatically, but for some reason it didn't. I also don't see a clear failure cause in the log that you shared.

In the meantime, you could try a workaround: install OpenSSL with Homebrew and instruct Ruby to link to that instead of the system one. Something like:

```sh
brew install openssl@3
RUBY_CONFIGURE_OPTS="--with-openssl-dir=$(brew --prefix openssl@3)" rbenv install 3.1.2
```
